unittest2.nim

```diff
@@ -357,12 +357,14 @@ method testStarted*(formatter: ConsoleOutputFormatter, testName: string) =
   formatter.isInTest = true
 
 method failureOccurred*(formatter: ConsoleOutputFormatter,
-                        checkpoints: seq[string], stackTrace: string) =
-  if stackTrace.len > 0:
-    echo stackTrace
+                        checkpoints: seq[string], stackTraces: seq[string]) =
+  # if stackTrace.len > 0:
+  #   echo stackTrace
   let prefix = if formatter.isInSuite: "    " else: ""
-  for msg in items(checkpoints):
+  for i, msg in checkpoints:
     echo prefix, msg
+    if stackTraces.len > 0:
+      echo stackTraces[i]
```

Comment:
This is not related to the error, but IMHO it is more intuitive to see first the exception and then the stacktrace.